cloudfour/cloudfour.com-patterns
# Generate JavaScript tokens

## Overview

This is an alternative approach to #788. Instead of exposing Theo tokens directly via Rollup, the tokens are preprocessed and JavaScript versions are generated.

The drawback to this approach are that the import paths are a little longer and recompiling token changes is a bit slower.

But there are a few benefits:

1. It appears to play nicely with TypeScript, and does not seem to require any separate declaration files or additions to our existing `types.ds` file.
1. It allows you to `import` values directly instead of as a glob.
1. In the future, this may invalidate our need for the Theo Webpack loader, which would further simplify our configuration.

## Screenshots

<img width="1079" alt="Screen Shot 2020-06-11 at 1 48 48 PM" src="https://user-images.githubusercontent.com/69633/84437921-a65c8680-abea-11ea-893f-e33b904912dc.png">

## Testing

1. Check out branch locally.
1. Attempt to import a design token of your choosing using one of those documented in Storybook. (To avoid having to create a new file, try adding it to the Elastic Textarea script.)
1. Run `npm run build`
1. See if the token made it to `dist/cloudfour-patterns.js`.
1. Run `npm run lint` and `npm run type` to make sure there are no errors after that change was made.

---

- Resolves #787

/CC @calebeby
